Core/GDCore/IDE/Events/EventsRefactorer.cpp

```diff
@@ -295,183 +296,114 @@ class GD_CORE_API ExpressionObjectFinder : public ExpressionParser2NodeWorker {
-                                             const gd::ProjectScopedContainers& projectScopedContainers,
-                                             gd::InstructionsList& actions,
-                                             gd::String oldName,
-                                             gd::String newName) {
-  bool somethingModified = false;
-
-  for (std::size_t aId = 0; aId < actions.size(); ++aId) {
-    const gd::InstructionMetadata& instrInfos =
-        MetadataProvider::GetActionMetadata(platform, actions[aId].GetType());
-    for (std::size_t pNb = 0; pNb < instrInfos.parameters.GetParametersCount(); ++pNb) {
-      // Replace object's name in parameters
-      if (gd::ParameterMetadata::IsObject(instrInfos.parameters.GetParameter(pNb).GetType()) &&
-          actions[aId].GetParameter(pNb).GetPlainString() == oldName)
-        actions[aId].SetParameter(pNb, gd::Expression(newName));
-      // Replace object's name in expressions
-      else if (ParameterMetadata::IsExpression(
-                   "number", instrInfos.parameters.GetParameter(pNb).GetType())) {
-        auto node = actions[aId].GetParameter(pNb).GetRootNode();
-
-        if (ExpressionObjectRenamer::Rename(platform, projectScopedContainers, "number", *node, oldName, newName)) {
-          actions[aId].SetParameter(
-              pNb, ExpressionParser2NodePrinter::PrintNode(*node));
-        }
-      }
-      // Replace object's name in text expressions
-      else if (ParameterMetadata::IsExpression(
-                   "string", instrInfos.parameters.GetParameter(pNb).GetType())) {
-        auto node = actions[aId].GetParameter(pNb).GetRootNode();
-
-        if (ExpressionObjectRenamer::Rename(platform, projectScopedContainers, "string", *node, oldName, newName)) {
-          actions[aId].SetParameter(
-              pNb, ExpressionParser2NodePrinter::PrintNode(*node));
-        }
-      }
-    }
+/**
+ * \brief Replace in expressions and in parameters of actions or conditions,
+ * references to the name of an object by another.
+ *
+ * \ingroup IDE
+ */
+class GD_CORE_API EventsObjectReplacer
+    : public ArbitraryEventsWorkerWithContext {
+public:
+  EventsObjectReplacer(const gd::Platform &platform_,
+                       const gd::ObjectsContainer &targetedObjectsContainer_,
+                       const gd::String &oldObjectName_,
+                       const gd::String &newObjectName_)
+      : platform(platform_),
+        targetedObjectsContainer(targetedObjectsContainer_),
+        oldObjectName(oldObjectName_), newObjectName(newObjectName_){};
+
+  virtual ~EventsObjectReplacer() {}
+
+private:
+  bool DoVisitInstruction(gd::Instruction &instruction,
+                          bool isCondition) override {
+    if (&targetedObjectsContainer !=
+        GetProjectScopedContainers()
+            .GetObjectsContainersList()
+            .GetObjectsContainerFromObjectName(oldObjectName)) {
+      return false;
+    }

```

Comment:
This check is probably useless and asking the `targetedObjectsContainer` to callers complicate things. Maybe I shouldn't have added it.

Reply:
Might be useless because of shadowing of names by more local containers. But who knows, might actually avoid wrongly renaming an object in case multiple containers have an object with the same name